dashboard/lib/announcements.rb

```diff
@@ -0,0 +1,60 @@
+class Announcements
+  @@announcements_data = nil
+  @@loaded = false
+  @@load_error = false
+  @@json_path = File.join("#{pegasus_dir}/sites.v3/code.org", 'announcements.json')
+
+  # enables unit tests
+  def self.set_file_path(path)
+    @@json_path = path
+    @@loaded = false
+    @@load_error = false
+  end
```

Comment:
Interested in a discussion about the tradeoff of this singleton approach with a method to enable testing, vs a non-singleton that optionally takes a file path on construction and refers to a static cache keyed by file path.